library/common/http/header_utility.cc

```diff
@@ -26,7 +26,7 @@ RequestHeaderMapPtr toRequestHeaders(envoy_headers headers) {
   auto transformed_headers = RequestHeaderMapImpl::create();
   transformed_headers->setFormatter(
       std::make_unique<
-          Extensions::Http::HeaderFormatters::PreserveCase::PreserveCaseHeaderFormatter>());
+          Extensions::Http::HeaderFormatters::PreserveCase::PreserveCaseHeaderFormatter>(false));
```

Comment:
To account for https://github.com/envoyproxy/envoy/pull/18997

Considering it's designed to be off by default, I imagine that we can disable it for our use cases until we have reason to do otherwise.